unit_testing/test_reports.py

```diff
@@ -3,38 +3,46 @@
 # pytest unit tests for spinalcordtoolbox.reports
 
 import pytest
+from pytest_cases import parametrize_with_cases
 import numpy as np
 
 from spinalcordtoolbox.image import Image
 from spinalcordtoolbox.reports.slice import Sagittal
+from spinalcordtoolbox.utils.sys import sct_test_path
 
 
-def labeled_data_test_params(path_in='sct_testing_data/t2/t2.nii.gz',
-                             path_seg='sct_testing_data/t2/labels.nii.gz'):
-    """Generate image/label pairs for various test cases of
-    test_sagittal_slice_get_center_spit."""
-    im_in = Image(path_in)            # Base anatomical image
-    im_seg_labeled = Image(path_seg)  # Base labeled segmentation
-    assert np.count_nonzero(im_seg_labeled.data) >= 2, "Labeled segmentation image has fewer than 2 labels"
+def im_base(path_in=sct_test_path('t2', 't2.nii.gz')):
+    # Base anatomical image
+    return Image(path_in)
```

Comment:
We're a bit stuck... I think we actually _do_ need something like `pytest_sessionstart` after all.

Reply:
New idea: Use a lock file to prevent multiple workers from fighting over test data downloading.

This is the approach recommended by [`pytest-xdist`](https://pypi.org/project/pytest-xdist/): 

> While pytest-xdist does not have a builtin support for ensuring a session-scoped fixture is executed exactly once, this can be achieved by using a lock file for inter-process communication.
>
> [Example]
>
> The example above can also be use in cases a fixture needs to execute exactly once per test session, like initializing a database service and populating initial tables.

This mentions fixtures, but I imagine this can be applied to session hooks as well.

Reply:
You avoid this if you mark this with `@fixture` -- and use it for the data instead of `sessionstart`. My understanding is that pytest behaves a lot like `make`: its collection phase is where it scans all the tests and resolves *what fixtures depend on what other fixtures*, then it creates all the fixtures in order -- there's not necessarily a single right order, it's a partial order, a dependency lattice -- and finally it actually executes all the tests (which, in `make`, would be like having a final `.PHONY: tests` target). In order to benefit from this dependency system we have to be using the fixtures system.

Reply:
In fact, we *should* be able to pass the `sct_test_data` fixture as a parameter to the fixtures that need it. We could make it work like how pytest's built in `tmp_dir` fixture works: it does some work before returning a path to the dependee.